src/flask/sansio/blueprints.py

```diff
@@ -353,11 +353,11 @@ def register(self, app: App, options: dict) -> None:
 
             if bp_subdomain is None:
                 bp_subdomain = blueprint.subdomain
-
-            if state.subdomain is not None and bp_subdomain is not None:
-                bp_options["subdomain"] = bp_subdomain + "." + state.subdomain
-            elif bp_subdomain is not None:
-                bp_options["subdomain"] = bp_subdomain
+            if bp_subdomain is not None:
+                if state.subdomain is not None:
+                    bp_options["subdomain"] = bp_subdomain + "." + state.subdomain
+                else:
+                    bp_options["subdomain"] = bp_subdomain
             elif state.subdomain is not None:
                 bp_options["subdomain"] = state.subdomain
```

Comment:
Those two lines are never executed since the condition if from the line 357 is always true. This should addressed before this pull request is accepted.